packages/ra-core/src/form/FormWithRedirect.tsx

```diff
@@ -70,6 +70,10 @@ const FormWithRedirect = ({
     const redirect = useRef(props.redirect);
     const onSave = useRef(save);
     const formGroups = useRef<{ [key: string]: string[] }>({});
+    const finalMutators =
```

Comment:
This might have a performance impact as we may create a new object every time, right?